crypto/crypto/src/ed25519.rs

```diff
@@ -76,6 +77,24 @@ impl Clone for Ed25519PrivateKey {
 #[derive(DeserializeKey, Clone, SerializeKey)]
 pub struct Ed25519PublicKey(ed25519_dalek::PublicKey);
 
+/// A private tag type that tracks if a public key is checked to protect against invalid point
+/// attacks, small subgroup attacks, and typos. This tag type is only used at compilation time.
+/// This type should remain private.
+#[cfg(feature = "const_generics")]
+type ValidPublicKeyTag = ValidPublicKeyTagKind<VALID_PUBLIC_KEY_TAG_MASK>;
+
+#[cfg(not(feature = "const_generics"))]
+struct ValidPublicKeyTag {}
```

Comment:
That's an OK name, but I'd prefer something that grammatically conveys the "has-happened-before" concept, hence `ValidatedPublicKeyTag`